source/common/network/connection_impl.cc

```diff
@@ -179,6 +179,13 @@ void ConnectionImpl::noDelay(bool enable) {
   // Set NODELAY
   int new_value = enable;
   rc = setsockopt(fd_, IPPROTO_TCP, TCP_NODELAY, &new_value, sizeof(new_value));
+#ifdef __APPLE__
+  if (-1 == rc && errno == EINVAL) {
```

Comment:
This seems like an OS X bug to me, but I digress. Hmm. In this case I would be in favor of just putting a comment in more of a comment on why this is done/breaks and maybe a TODO?